File: examples/02_STT/node/stt_watson_json.js
//Example STT - "watson JSON"
// npm install request@2.81.0

var fs = require('fs');
var request = require('request');

function listen(something){
  var APIkey = "YOUR_API_KEY";
  var url = "YOUR_API_URL";
  var endpoint = "/v1/recognize?timestamps=true&max_alternatives=3";
  var auth = 'apikey:' + APIkey;
  var data =  Buffer.from(fs.readFileSync(something));
  var options = {
      method: 'POST',
      headers: {'Content-Type': 'audio/flac', 'Authorization': 'Basic ' + Buffer.from(auth).toString('base64')},
      url: url+endpoint,
      body: data
  };
  request.post(options, function (error, response, body) {
    fs.appendFileSync('watson.json', body);
    console.log("Saved!")
  });
}

listen("_audio/audio-file.flac");

File: examples/02_STT/python/stt_watson_json.py
# Example STT - "watson JSON"
# --- Python 2.x ---
# sudo pip install requests
# --- Python 3.x ---
# python3 -m venv .venv
# source .venv/bin/activate
# pip install requests

import sys
import base64
import requests

def listen(something):
    APIkey = 'YOUR_API_KEY'
    url = 'YOUR_API_URL'
    endpoint = '/v1/recognize?timestamps=true&max_alternatives=3'
    data = open(something, 'rb').read()
    auth = 'apikey:' + APIkey
    headers = {'Content-Type': 'audio/flac', 'Authorization': 'Basic ' + base64.b64encode(
        auth.encode()).decode()}
    r = requests.post(headers=headers, url=url+endpoint, data=data)
    with open("watson.json", "a") as myfile:
        myfile.write(r.text)
        print("Saved!")


listen("_audio/audio-file.flac")
